tensorboard/components/tf_data_selector/tf-data-selector.ts

```diff
@@ -207,8 +209,19 @@ Polymer({
     const {runs, tagRegex} = event.detail;
     const experiment = event.target.experiment;
     const expId = experiment.id != null ? experiment.id : NO_EXPERIMENT_ID;
+
+    // Ignore events from an experiment that is removed.
+    if (expId != NO_EXPERIMENT_ID && !this._experimentIds.includes(expId)) {
+      return;
+    }
+
     const newSelections = new Map(this._selections);
     newSelections.set(expId, {experiment, runs, tagRegex});
+
+    // Ignore the selection changed event if it makes no tangible difference to
+    // the _selections.
```

Comment:
Maybe elaborate here to document why we might see such an event - i.e. due to async behavior when an experiment gets removed?

Reply:
Done! Thanks for the feedback